src/bp/core/services/bot-service.ts

```diff
@@ -138,6 +141,44 @@ export class BotService {
     }
   }
 
+  async exportBot(botId: string): Promise<Buffer> {
+    return this.ghostService.forBot(botId).exportToArchiveBuffer()
+  }
+
+  async importBot(botId: string, archive: Buffer, allowOverwrite?: boolean): Promise<void> {
+    const alreadyExists = (await this.getBotsIds()).includes(botId)
+    if (alreadyExists) {
+      if (!allowOverwrite) {
+        return this.logger.error(`Cannot import the bot ${botId}, it already exists, and overwrite is not allowed`)
+      } else {
+        this.logger.warn(`The bot ${botId} already exists, files in the archive will overwrite existing ones`)
+      }
+    }
+    const tmpDir = tmp.dirSync({ unsafeCleanup: true })
+    const tmpFolder = tmpDir.name
+
+    try {
+      await extractArchive(archive, tmpFolder)
+      const api = await createForGlobalHooks()
+
+      const hookResult = {
+        allowImport: true
+      }
+
+      await this.hookService.executeHook(new Hooks.BeforeBotImport(api, botId, tmpFolder, hookResult))
```

Comment:
It would be awesome if a hook could return a value instead of altering the object passed in param. I imagine that is not possible because we're using NodeVM ?

Reply:
not possible at the moment